crates/wash-cli/src/cmd/dev/session.rs

```diff
@@ -431,4 +403,64 @@ impl WashDevSession {
 
         Ok((nats_child, wadm_child, wasmcloud_child))
     }
+
+    async fn get_host_id(log_path: PathBuf) -> anyhow::Result<String> {
+        let log_file = tokio::fs::File::open(&log_path)
+            .await
+            .with_context(|| format!("failed to open log file @ [{}]", &log_path.display()))?;
+
+        // looks for the two variations of the log line containing the host_id:
+        //   JSON: "host_id":"ABC123"
+        //   LOG:  host_id="ABC123"
+        let re = regex::Regex::new(r#"(?:\"host_id\":\s?\"|host_id=\")([A-Z0-9]+)\""#)
```

Comment:
merged the regex so it doesn't depend on the cli options. It just works against std log + structured json.